lib/authorizations.js

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) 2019-2023 Digital Bazaar, Inc. All rights reserved.
+ */
+import * as bedrock from '@bedrock/core';
+import {httpClient} from '@digitalbazaar/http-client';
+
+const {config, util: {BedrockError}} = bedrock;
+
+export async function authorizeRegistration({req}) {
+  const cfg = config['account-http'];
+  const {authorizationRequired} = cfg.registration;
+  const {authorization} = req.body;
+  if(!(authorization?.type === authorizationRequired &&
+      authorization.token)) {
+    throw new BedrockError(
+      `Authorization using "${authorizationRequired}" is required.`, {
+        name: 'NotAllowedError',
+        details: {
+          httpStatusCode: 403,
+          public: true
+        }
+      });
+  }
+
+  const remoteIp = req.socket.remoteAddress;
```

Comment:
My understanding is that this won't work in deployments, and that we need to do something like this pattern:

https://github.com/digitalbazaar/bedrock-kms-http/blob/f4fdfe48c55307b04290c607e42484a9a03f7ee6/lib/helpers.js#L24

cc: @mattcollier

Reply:
Updated, the goal is to get the end user's IP address.